source/distributed/grid_refinement.cc

```diff
@@ -519,45 +582,43 @@ namespace parallel
         Assert(criteria.is_non_negative(),
                dealii::GridRefinement::ExcNegativeCriteria());
 
-        // first extract from the vector of indicators the ones that correspond
-        // to cells that we locally own
-        Vector<Number> locally_owned_indicators(
-          tria.n_locally_owned_active_cells());
-        get_locally_owned_indicators(tria, criteria, locally_owned_indicators);
-
-        MPI_Comm mpi_communicator = tria.get_communicator();
-
-        // figure out the global max and min of the indicators. we don't need it
-        // here, but it's a collective communication call
-        const std::pair<double, double> global_min_and_max =
-          dealii::internal::parallel::distributed::GridRefinement::
-            compute_global_min_and_max_at_root(locally_owned_indicators,
-                                               mpi_communicator);
-
-        const double total_error =
-          compute_global_sum(locally_owned_indicators, mpi_communicator);
-        double top_threshold, bottom_threshold;
-        top_threshold = dealii::internal::parallel::distributed::
-          GridRefinement::RefineAndCoarsenFixedFraction::compute_threshold(
-            locally_owned_indicators,
-            global_min_and_max,
-            top_fraction_of_error * total_error,
-            mpi_communicator);
-
-        // compute bottom threshold only if necessary. otherwise use the lowest
-        // threshold possible
-        if (bottom_fraction_of_error > 0)
-          bottom_threshold = dealii::internal::parallel::distributed::
-            GridRefinement::RefineAndCoarsenFixedFraction::compute_threshold(
-              locally_owned_indicators,
-              global_min_and_max,
-              (1. - bottom_fraction_of_error) * total_error,
-              mpi_communicator);
-        else
-          bottom_threshold = std::numeric_limits<Number>::lowest();
+        switch (norm_type)
+          {
+            case VectorTools::NormType::L1_norm:
+              // evaluate norms on subsets and compare them as
+              //   c_0 + c_1 + ... < fraction * l1-norm(c)
+              refine_and_coarsen_fixed_fraction_l1_norm(
+                tria,
+                criteria,
+                top_fraction_of_error,
+                bottom_fraction_of_error);
+              break;
+
+            case VectorTools::NormType::L2_norm:
+              {
+                // we do not want to evaluate norms on subsets as
+                //   sqrt(c_0^2 + c_1^2 + ...) < fraction * l2-norm(c)
+                // square fractions and criteria instead and work with
```

Comment:
I don't think I understand what you want to say with "square fractions and criteria"